### PR TITLE
feat(CI): optimize rust deps for Docker CI

### LINKED ISSUE #1265: feat(CI): Optimize rust builds
Our new CI doesn't cache anything on the rust build side: https://github.com/yetanotherco/aligned_layer/pull/1044

### PR DESCRIPTION
**Pull request by samoht9277**
# Optimization for rust builds inside the CI

## Description
This PR aims to improve build times by caching dependencies.

To test this, simply run the `Send proofs to network / Test network with Docker Compose` GitHub workflow from this branch. If no caché is found, it will compile all, like it has been always doing. If caché is found, it will skip through the cached steps, improving time overall.

## Type of change

Please delete options that are not relevant.

- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Refactor

## Checklist

- [ ] “Hotfix” to `testnet`, everything else to `staging`
- [x] Linked to Github Issue -> https://github.com/yetanotherco/aligned_layer/issues/1265
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [x] This change is an Optimization
  - [x] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]() 
- [ ] If your PR changes the Operator compatibility (Ex: Upgrade prover versions)
  - [ ] This PR adds compatibility for operator for both versions and do not change batcher/docs/examples
  - [ ] This PR updates batcher and docs/examples to the newer version. This requires the operator are already updated to be compatible

### CODE REVIEW

File: .github/workflows/send-proofs-docker.yml [hunk 1]
```diff
@@ -53,4 +53,4 @@ jobs:
 
     - name: Ensure admin permissions in _work
       if: always()
-      run: sudo chown admin:admin -R /home/admin/actions-runner/_work/
+      run: sudo chown admin:admin -R /home/admin/actions-runner/_work/
```
Comment (klaus993) [inline .github/workflows/send-proofs-docker.yml]: fix no newline at the end of the file

(you can configure VIM to do this automatically)

File: config-files/config-aggregator-docker.yaml [hunk 2]
```diff
@@ -29,4 +29,4 @@ aggregator:
   telemetry_ip_port_address: localhost:4001
   garbage_collector_period: 2m #The period of the GC process. Suggested value for Prod: '168h' (7 days)
   garbage_collector_tasks_age: 20 #The age of tasks that will be removed by the GC, in blocks. Suggested value for prod: '216000' (30 days)
-  garbage_collector_tasks_interval: 10 #The interval of queried blocks to get an old batch. Suggested value for prod: '900' (3 hours)
+  garbage_collector_tasks_interval: 10 #The interval of queried blocks to get an old batch. Suggested value for prod: '900' (3 hours)
```
Comment (klaus993) [inline config-files/config-aggregator-docker.yaml:32]: fix no newline at the end of the file

File: docker/aggregator.Dockerfile [hunk 3]
```diff
@@ -9,6 +9,7 @@ COPY core         ./core
 COPY metrics      ./metrics
 COPY contracts/bindings/ ./contracts/bindings
 
+RUN go get github.com/ethereum/go-ethereum@latest
```
Comment (klaus993) [inline docker/aggregator.Dockerfile:12]: why do we need this?

File: docker/aligned_base.Dockerfile [hunk 4]
```diff
@@ -31,12 +28,83 @@ RUN go install github.com/maoueh/zap-pretty@latest
...
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
 WORKDIR /aligned_layer
 
 COPY Makefile .
-COPY operator ./operator
-COPY batcher/aligned-sdk ./batcher/aligned-sdk
+#COPY operator ./operator
+#COPY batcher/aligned-sdk ./batcher/aligned-sdk
```
Comment (klaus993) [inline docker/aligned_base.Dockerfile]: why is this commented?

File: docker/aggregator.Dockerfile (hunk 3 above)
Comment (samoht9277) [reply docker/aggregator.Dockerfile:12]: I was running some tests. This ended up not being needed. It's already removed.

File: docker/aligned_base.Dockerfile (hunk 4 above)
Comment (samoht9277) [reply docker/aligned_base.Dockerfile]: Same as above. I was testing stuff and now it is no longer needed to be commented.

File: docker/aggregator.Dockerfile (hunk 3 above)
Comment (samoht9277) [reply docker/aggregator.Dockerfile:12]: Actually, we do need this. I tried removing this and it breaks.


```
#15 5.504 /root/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.0/crypto/signature_nocgo.go:91:14: assignment mismatch: 2 variables but btc_ecdsa.SignCompact returns 1 value
#15 ERROR: process "/bin/sh -c go build -o ./aligned-layer-aggregator aggregator/cmd/main.go" did not complete successfully: exit code: 1
------
 > [aggregator builder 9/9] RUN go build -o ./aligned-layer-aggregator aggregator/cmd/main.go:
0.668 go: downloading github.com/prometheus/procfs v0.13.0
0.792 go: downloading github.com/consensys/gnark-crypto v0.12.2-0.20240215234832-d72fcb379d3e
0.794 go: downloading github.com/shurcooL/graphql v0.0.0-20230722043721-ed46e5a46466
0.881 go: downloading golang.org/x/sys v0.19.0
0.905 go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.3
3.885 go: downloading github.com/gorilla/websocket v1.5.1
3.885 go: downloading github.com/shirou/gopsutil v3.21.6+incompatible
3.897 go: downloading github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
5.504 # github.com/ethereum/go-ethereum/crypto
5.504 /root/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.0/crypto/signature_nocgo.go:91:14: assignment mismatch: 2 variables but btc_ecdsa.SignCompact returns 1 value
------
failed to solve: process "/bin/sh -c go build -o ./aligned-layer-aggregator aggregator/cmd/main.go" did not complete successfully: exit code: 1
make: *** [Makefile:780: docker_build] Error 17
```

**Review by MauroToscano — APPROVED** [review]

**Review by JuArce — APPROVED** [review]

**Pull request merged**